### PR TITLE
Member creation function.

### PR DESCRIPTION
**Pull request by rahulgi**
Create a Firebase function that "applies" new REQUEST_INVITE operations to the `members` view. The member creation function is triggered by the creation of a new Operation object in the database.

### CODE REVIEW

**Comment by rahulgi** [comment]
This PR is pretty noisy in terms of commits - perhaps we should do a squash+rebase instead of pure rebase when merging to master.

File: functions/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    });
+}
+
+exports.onCreateOperation = functions.firestore
+    .document('/operations/{operationId}')
+    .onCreate((event) => {
+        const newOperation = event.data.data();
+        switch(newOperation.op_code) {
+            case REQUEST_INVITE:
+                createMemberFromOperation(newOperation).then(() => {
```
Comment (markulrich) [inline functions/index.js]: Does `try/except` surrounding `await` work?

File: functions/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    const {
+        // unused to_uid,
+        // unused to_mid,
+        video_url,
+        full_name
+    } = data;
+
+    const newMember = db.collection('members').doc(creator_uid);
+    // By using `create`, we ensure that this operation will fail
```
Comment (markulrich) [inline functions/index.js]: This also means also means that if they make some typo mistake during invite process then realize it once they get to their profile page (or later decide to request from someone else if original person does not confirm cause they are run over by bus), cannot instantly go back and change it. What if we instead allow people to update only if their `invite_confirmed` is still false? For consistency I think we would have to use `db.runTransaction`, first grab existing user (if it exists) then update (if safe to do so)

File: functions/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+exports.onCreateOperation = functions.firestore
+    .document('/operations/{operationId}')
+    .onCreate((event) => {
+        const newOperation = event.data.data();
+        switch(newOperation.op_code) {
+            case REQUEST_INVITE:
+                createMemberFromOperation(newOperation).then(() => {
+                    return event.data.ref.set({
```
Comment (markulrich) [inline functions/index.js]: Same effect but `.update` seems more semantically appropriate here https://stackoverflow.com/questions/46597327/difference-between-set-with-merge-true-and-update ?

File: functions/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        block_seq,
+        created_at,
+        creator_uid,
+        creator_mid,
+        data,
+        op_code,
+        op_seq
+    } = operation;
+
+    if (applied || op_code !== REQUEST_INVITE) {
```
Comment (markulrich) [inline functions/index.js]: Need more data validation as at this point our `firestore.rules` are quite simple. We should check every field coming from a (possibly malicious) client. Eg that block_at block_seq op_seq are all null, that created_at really is a date, that full_name and creator_mid are strings and they match, etc

File: functions/index.js (hunk 1 above)
Comment (rahulgi) [reply functions/index.js]: I think async-await isn't supported by the version of node firestore runs.

File: functions/index.js (hunk 2 above)
Comment (rahulgi) [reply functions/index.js]: After discussion - let's add functionality for someone to update their information from their profile page after requesting an invite - it seems like an odd flow for people to be able to request multiple invites from a single account.

File: functions/index.js (hunk 3 above)
Comment (rahulgi) [reply functions/index.js]: 👍

File: functions/index.js (hunk 4 above)
Comment (rahulgi) [reply functions/index.js]: After discussion: needs more thought, see issue #61.

**Pull request merged**